packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,151 @@
+ */
+export interface CustomLambdaDeploymentConfigProps {
+
+  /**
+   * The type of deployment config, either CANARY or LINEAR
+   */
+  readonly type: CustomLambdaDeploymentConfigType;
+
+  /**
+   * The integer percentage of traffic to shift:
+   * - For LINEAR, the percentage to shift every interval
+   * - For CANARY, the percentage to shift until the interval passes, before the full deployment
+   */
+  readonly percentage: number;
+
+  /**
+   * The interval, in number of minutes:
+   * - For LINEAR, how frequently additional traffic is shifted
+   * - For CANARY, how long to shift traffic before the full deployment
+   */
+  readonly interval: Duration;
+}
+
+/**
+ * A custom Deployment Configuration for a Lambda Deployment Group.
+ *
+ */
+export class CustomLambdaDeploymentConfig extends Construct implements ILambdaDeploymentConfig {
+
+  /**
+   * The name of the deployment config
+   */
+  public readonly deploymentConfigName: string;
+
+  /**
+   * The arn of the deployment config
+   */
+  public readonly deploymentConfigArn: string;
+
+  public constructor(scope: Construct, id: string, props: CustomLambdaDeploymentConfigProps) {
+    super(scope, id);
+    this.validateParameters(props);
+
+    const deploymentType = 'TimeBased' + props.type.toString();
+    const intervalMinutes = props.interval.toMinutes().toString();
+    const percentage = props.percentage.toString();
+
+    let routingConfig;
+    let customDeploymentConfigName;
+    if (props.type == CustomLambdaDeploymentConfigType.CANARY) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedCanary: {
+          canaryInterval: intervalMinutes,
+          canaryPercentage: percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaCanary' + percentage+ 'Percent' + intervalMinutes + 'Minutes';
+    } else if (props.type == CustomLambdaDeploymentConfigType.LINEAR) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedLinear: {
+          linearInterval: intervalMinutes,
+          linearPercentage: percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaLinear' + percentage + 'Percent' + intervalMinutes + 'Minutes';
+    }
+
+    // What the config is called in the AWS console
+    // We tack the id on so that, if the user has multiple configs with identical settings,
+    // deleting one will not affect the other
+    const resourceName = id + customDeploymentConfigName;
+    this.deploymentConfigName = resourceName;
+    this.deploymentConfigArn = arnForDeploymentConfig(resourceName);
+
+    // The AWS Custom Resource that calls CodeDeploy to create and delete the resource
+    new AwsCustomResource(this, 'DeploymentConfig', {
+      onCreate: { // Run on creation only, to make the resource
+        service: 'CodeDeploy',
+        action: 'createDeploymentConfig',
+        parameters: {
+          deploymentConfigName: resourceName,
```

Comment:
Should we actually allow the customer to pass this name here? Of course, it should be optional, and the default should be for the name to be auto-generated.

Reply:
Added